framework/behaviors/SluggableBehavior.php

```diff
@@ -171,6 +171,10 @@ protected function isNewSlugNeeded()
             return false;
         }
 
+        if ($this->attribute === null) {
```

Comment:
Seems like we can optimize this condition for the case when `$value` is a string. We can compare it to the current slug value. Do you think it worth implementing?

For the case when `$value` is `callable` the comment would be:
> Callback-based non-immutable slugs should be recalculated each time.